src/Eshopworld.WorkerProcess/LeaseAllocator.cs

```diff
@@ -34,10 +34,10 @@ public async Task<ILease> AllocateLeaseAsync(Guid instanceId)
         {
             _telemetry.Publish(new LeaseAcquisitionEvent(instanceId, _options.Value.WorkerType,
                 _options.Value.Priority, $"Lease allocation started for instance {instanceId}"));
-           
+
             var lease = await _leaseStore.ReadByLeaseTypeAsync(_options.Value.WorkerType).ConfigureAwait(false);
             if (lease?.LeasedUntil != null && lease.LeasedUntil.Value > ServerDateTime.UtcNow)
-                return null;
+                return lease.InstanceId == instanceId ? lease : null;
```

Comment:
Would it make sense to add a telemetry event to notify we are reusing instance

Reply:
I guess you are right, let me add that.

Reply:
cool thanks!